src/kaleidoscope/driver/keyscanner/None.h

```diff
@@ -0,0 +1,40 @@
+/* -*- mode: c++ -*-
+ * kaleidoscope::driver::keyscanner::None -- Dummy keyscanner component
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <Arduino.h>
+
+#include "kaleidoscope/driver/keyscanner/Base.h"
+
+namespace kaleidoscope {
+namespace driver {
+namespace keyscanner {
+
+struct NoProps : kaleidoscope::driver::keyscanner::BaseProps {};
```

Comment:
Would Base::Props instead of BaseProps make sense, basically declaring that ::Props is always the properties struct inside one of our drivers?

Reply:
That sounds reasonable to me. Will see how the compiler likes it.

Reply:
On second thought: no, can't do. We'd get a hen & egg problem then with derived classes.

for example, `keyscanner::No` would need the props specified for its parent earlier than we can decleare `keyscanner::No::Props` (which would be inside the class). So we'd still have a `NoProps` outside, to be able to write `keyscanner::Base<NoProps>` (as `keyscanner::Base<No::Props>` doesn't work, due to `No::Props` not being declared yet).